utility/SPIDEV/spi.cpp

```diff
@@ -99,39 +100,21 @@ void SPI::init()
 	}
 }
 
-uint8_t SPI::transfer(uint8_t tx_)
+uint8_t SPI::transfer(uint8_t tx)
 {
     pthread_mutex_lock (&spiMutex);
-	int ret;
-  	uint8_t tx[1] = {tx_};
-	uint8_t rx[1];
-     
-    this->init();
-	struct spi_ioc_transfer tr = {
-	tr.tx_buf = (unsigned long)&tx[0],
-	tr.rx_buf = (unsigned long)&rx[0],
-	tr.len = 1,//ARRAY_SIZE(tx),
-	tr.delay_usecs = 0,
-	tr.cs_change=1,
-	tr.bits_per_word = this->bits,
-	};
-	
-    tr.speed_hz = this->speed,	
-	//Note: On RPi, for some reason I started getting 'bad message' errors, and changing the struct as below fixed it, until an update...??
-	//
-	/*	// One byte is transfered at once
-
-
-	uint8_t rx[ARRAY_SIZE(tx)] = {0};
-	struct spi_ioc_transfer tr;
-	tr.tx_buf = (unsigned long)tx;
-	tr.rx_buf = (unsigned long)rx;
-	tr.len = ARRAY_SIZE(tx);
-	tr.delay_usecs = 0;
-	tr.cs_change = 1;
-	tr.speed_hz = this->speed;
-	tr.bits_per_word = this->bits;*/
 
+    struct spi_ioc_transfer tr;
+    memset(&tr, 0, sizeof(tr));
+    tr.tx_buf = (unsigned long)&tx;
+    tr.rx_buf = (unsigned long)&tx;
```

Comment:
The use of tx for output _and_ input buffers is innecessarily grubby - why not leave rx as a separate entity?